wgsl/index.bs

```diff
@@ -689,84 +682,244 @@ An 8-bit byte is the most basic unit of [=host-shareable=] memory.
+    };
+
+    [[group(0), binding(0)]]
+    var<storage> storage_buffer : B;
+  </xmp>
+</div>
+
+<div class='example wgsl' heading='Layout of structures with explicit member strides and alignments'>
+  <xmp highlight='rust'>
+    [[block]] struct A {                           //             align(8)  size(32)
+        u : f32;                                   // offset(0)   align(4)  size(4)
+        v : f32;                                   // offset(4)   align(4)  size(4)
+        w : vec2<f32>;                             // offset(8)   align(8)  size(8)
+       [[size(16)]] x: f32;                        // offset(16)  align(4)  size(16)
+    };
+
+    [[block]] struct B {                           //             align(16) size(208)
+        a : vec2<f32>;                             // offset(0)   align(8)  size(8)
+        // -- implicit member alignment padding -- // offset(8)   align(1)  size(8)
+        b : vec3<f32>;                             // offset(16)  align(16) size(12)
+        c : f32;                                   // offset(28)  align(4)  size(4)
+        d : f32;                                   // offset(32)  align(4)  size(4)
+        // -- implicit member alignment padding -- // offset(36)  align(1)  size(12)
+        [[align(16)]] e : A;                       // offset(48)  align(16) size(32)
+        f : vec3<f32>;                             // offset(80)  align(16) size(12)
+        // -- implicit member alignment padding -- // offset(92)  align(1)  size(4)
+        g : [[stride(32)]] array<A, 3>;            // offset(96)  align(8)  size(96)
+        h : i32;                                   // offset(192) align(4)  size(4)
+        // -- implicit struct size padding --      // offset(196) align(1)  size(12)
+    };
+
+    [[group(0), binding(0)]]
+    var<uniform> uniform_buffer : B;
+  </xmp>
+</div>
+
+#### Array Layout Rules ####  {#array-layout-rules}
+
+An array element stride is the number of bytes from the start of one array
+element to the start of the next element.
+
+The first array element always has a zero byte offset from the start of the
+array.
+
+If the array type is annotated with an explicit [=stride=] decoration then this
+will be used as the array stride, otherwise the array uses an implicit stride
+equal to the size of the array's element type, rounded up to the alignment of
+the element type:
+
+<p algorithm="array implicit element stride">
+  [=StrideOf=](array<|T|[, |N|]>) = [=roundUp=]([=AlignOf=](T), [=SizeOf=](T))
+</p>
+
+<div class='example wgsl' heading='Implicit / explicit array strides'>
+  <xmp highlight='rust'>
+    // Array with an implicit element stride of 16 bytes
+    var implicit_stride : array<vec3<f32>, 8>;
+
+    // Array with an explicit element stride of 32 bytes
+    var explicit_stride : [[stride(32)]] array<vec3<f32>, 8>;
+  </xmp>
+</div>
+
+Array elements must not overlap. Arrays decorated with the [=stride=] attribute
```

Comment:
should this say that the stride has to be a *multiple* of the size?

Reply:
Done